### DISCUSSION TITLE
Nested select feature + video

### DISCUSSION THREAD

**Discussion by dantownsend** [piccolo-orm/piccolo#214] [General]
We recently added a cool new feature, which I'm calling 'nested select' because I can't think of a better way to describe it ...

```python
>>> Band.select(Band.all_columns(), Band.manager.all_columns()).output(nested=True).run_sync()
{'id': 1, 'name': 'Pythonistas', 'manager': {'id': 1, 'name': 'Guido'}}
```

It also has great Pydantic support.

Here's video about it:

https://www.youtube.com/watch?v=NRO0HyFCCCI

**Comment by aminalaee** [comment]
@dantownsend Do you want to drop a link to these video playlists in README/doc?

**Reply by sinisaos** [reply]
@aminalaee This is great idea. @dantownsend You can also update the tutorials on the main [Piccolo ORM website](https://piccolo-orm.com/tutorial/videos/)

**Reply by dantownsend** [reply]
Yes, good points - we should link to these videos in the docs and website. Will update the website now.

**Comment by wmshort** [comment]
Brilliant. The new nested model support is fantastic; exactly what was needed. Can it be extend also to ``objects()``?

**Reply by dantownsend** [reply]
It's possible - I'll have a think about how best to do it.

**Comment by dantownsend** [comment]
I've added a prototype for nested objects - any thoughts on the API are welcome:

https://github.com/piccolo-orm/piccolo/pull/217
